builtin/credential/aws-ec2/path_login.go

```diff
@@ -294,12 +324,17 @@ func (b *backend) pathLoginUpdate(
 		if err = validateMetadata(clientNonce, identityDoc.PendingTime, storedIdentity, roleEntry); err != nil {
 			return logical.ErrorResponse(err.Error()), nil
 		}
+
+		if !disallowReauthentication && storedIdentity.DisallowReauthentication {
```

Comment:
Can be simplified to `disallowReauthentication |= storedIdentity.DisallowReauthentication"`. Either way is fine!